source/Naturals/IdOrder.lagda

```diff
@@ -0,0 +1,400 @@
+negation proofs for the Naturals. The use of transport is ergonomic as
+each term is directly an identity term. Our type definition also has
+strong computational properties that assist Agda's case matching as
+opposed to the standard definition; you will never see it attempt to match
+for the case 'succ x ≼ zero' as Agda can automatically detect this type
+is uninhabited.
+
+We use the curly order '≼' so as not to conflict with the one already
+defined for the Naturals. Although the early portions of this module were
+written myself, after I translated Escardo's definition of ≤-induction to
+this type I decided to go further and have begun translating all the Order
+lemmas in the standard module to accommodate this definition. For the most
+part, I have attempted to preserve the titles and type definitions
+exactly, but in the earlier sections of the modules I took some liberties
+as I wanted to tailor the most fundamental lemmas to the character of this
+type, and also to make certain operations more ergonomic and reduce the
+character count of proofs. Notably, 'equal-gives-less-than-or-equal' is
+simply '≼-＝' here.
+
+Aside from the fundamental lemmas as well as ones concerning absolute
+difference the credit for most of the type definitions that will be listed
+here goes to the efforts of Martin Escardo, Tom de Jong, and Andrew Sneap.
+Sometimes I needed to tailor their proofs to fit our order definition, but
+in many cases I was able to use essentially the same principles from
+proofs in the other module to guide the equivalent proofs here.
+
+\begin{code}
+
+{-# OPTIONS --safe --without-K --exact-split #-}
+
+module Naturals.IdOrder where
+
+open import MLTT.Spartan renaming (_+_ to _∔_)
+open import MLTT.Plus-Properties using (+functor)
+
+open import Naturals.Addition using (_+_; zero-left-neutral; succ-left;
+ addition-commutativity; addition-right-cancellable)
+open import Naturals.AbsoluteDifference
+open import Naturals.Order using (_≤ℕ_)
+open import Naturals.Properties using (positive-not-zero; succ-lc)
+open import Naturals.Multiplication using (_*_)
+
+open import Notation.General
+open import Notation.Order
+open import NotionsOfDecidability.Decidable
+
+open import UF.Base using (transport₂)
+open import UF.DiscreteAndSeparated using (discrete-types-are-sets;
+ is-discrete; ℕ-is-set; props-are-discrete)
+open import UF.Equiv using (_≅_)
+open import UF.Sets using (is-set)
+open import UF.Subsingletons using (is-prop)
+
+_≼ℕ_ : ℕ → ℕ → 𝓤₀ ̇
+x ≼ℕ y = ∣ y - x ∣ + x ＝ y
+
+infix 30 _≼ℕ_
+
+instance
+ Curly-Order-ℕ-ℕ : Curly-Order ℕ ℕ
+ _≼_ {{Curly-Order-ℕ-ℕ}} = _≼ℕ_
+
+private
+ example : 0 ≼ 2
+ example = refl
+
+ example2 : ¬ (1 ≼ 0)
+ example2 = positive-not-zero 1
+
+≼-is-prop-valued : (x y : ℕ) → is-prop (x ≼ y)
+≼-is-prop-valued x y p q = ℕ-is-set p q
+
+≼-is-discrete : (x y : ℕ) → is-discrete (x ≼ y)
+≼-is-discrete x y = props-are-discrete (≼-is-prop-valued x y)
+
+≼-is-set : (x y : ℕ) → is-set (x ≼ y)
```

Comment:
I am not sure why you are proving `≼-is-discrete` and `≼-is-set`. For the latter you could use `props-are-sets`. Do you really need this? I would suggest to delete both.